### PR TITLE
Contentful/Several error reporting improvements

### PR DESCRIPTION
**Pull request by dpinol**
See commit messages

### CODE REVIEW

File: packages/botonic-plugin-contentful/src/cms/contents.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ export abstract class TopContent extends Content {
     return new TopContentId(this.contentType, this.id)
   }
 
-  cloneWithFollowUp(newFollowUp: FollowUp): TopContent {
+  cloneWithFollowUp(newFollowUp: FollowUp): this {
```
Comment (marcpdw) [inline packages/botonic-plugin-contentful/src/cms/contents.ts:72]: Why wasn't the return type correct before? `TopContent` is the right thing here, no?

File: packages/botonic-plugin-contentful/src/contentful/contents/button.ts [hunk 2]
```diff
@@ -32,7 +32,14 @@ export class ButtonDelivery {
   ): Promise<cms.Button> {
     // we could pass the entry to fromId to avoid fetching it again, but it makes
     // the code more complex when the reference is a button
-    return this.fromId(entry.sys.id, context)
+    try {
+      return await this.fromId(entry.sys.id, context)
```
Comment (marcpdw) [inline packages/botonic-plugin-contentful/src/contentful/contents/button.ts:36]: If the method is defined as `async`, is it necessary to `await` for the result? Is it double wrapping it into Promises?

File: packages/botonic-plugin-contentful/src/contentful/traverser.ts [hunk 3]
```diff
@@ -200,7 +200,7 @@ export class I18nEntryTraverser {
     if (
       !val.fields ||
       val.fields.payload ||
-      val.sys.contentType.sys.id == ButtonDelivery.BUTTON_CONTENT_TYPE
+      val.sys.contentType?.sys.id == ButtonDelivery.BUTTON_CONTENT_TYPE
```
Comment (marcpdw) [inline packages/botonic-plugin-contentful/src/contentful/traverser.ts:203]: I normally consider the `===` as good practise

File: packages/botonic-plugin-contentful/src/cms/contents.ts (hunk 1 above)
Comment (dpinol) [reply packages/botonic-plugin-contentful/src/cms/contents.ts:72]: it's explained here https://www.typescriptlang.org/docs/handbook/advanced-types.html#polymorphic-this-types

File: packages/botonic-plugin-contentful/src/contentful/contents/button.ts (hunk 2 above)
Comment (dpinol) [reply packages/botonic-plugin-contentful/src/contentful/contents/button.ts:36]: I only do the await when I use the result or then when want to treat the exceptions with a try/catch (through I'm not 100% about this one)

Comment (ericmarcos) [reply packages/botonic-plugin-contentful/src/contentful/contents/button.ts:36]: @dpinol is right here. `return await` makes sense when you want to catch the rejection and return other stuff (and it only makes sense in that situation).
See https://jakearchibald.com/2017/await-vs-return-vs-return-await/

**Pull request merged**